frontend/lib/types/CPtrType.cpp

```diff
@@ -36,6 +36,14 @@ const owned<CPtrType>& CPtrType::getCPtrType(Context* context,
   return QUERY_END(result);
 }
 
+const owned<CPtrType>& CPtrType::getCPtrConstType(Context* context,
+                                                  const CPtrType* instantiatedFrom,
+                                                  const Type* eltType) {
+  QUERY_BEGIN(getCPtrConstType, context, instantiatedFrom, eltType);
+  auto result = toOwned(new CPtrType(instantiatedFrom, eltType, /*isConst*/ true));
+  return QUERY_END(result);
+}
```

Comment:
Seems to me like the query can be a single function that takes a boolean argument. I'm not too bent on making it that way, though.